RHIDP-12595: Migrate RHDH UI features to New Frontend System (#2107)
redhat-developer/red-hat-developers-documentation-rhdh: modules/shared/proc-get-instant-help-with-an-ai-powered-chatbot.adoc

:_mod-docs-content-type: PROCEDURE

> [id="get-instant-help-with-an-ai-powered-chatbot_{context}"]

Reviewer: Just for consistency with the rest of the docs, I would suggest we say "virtual assistant" instead of "chatbot".
See: 
https://docs.redhat.com/en/documentation/red_hat_developer_hub/1.9/html/interacting_with_red_hat_developer_lightspeed_for_red_hat_developer_hub/index

= Get instant help with an AI-powered chatbot

[role="_abstract"]
Add an AI chatbot to {product} that answers questions, debugs issues, and guides you through tasks using natural language. The chatbot is accessible through a floating action button that opens an app drawer.

.Prerequisites

* You have installed the `@red-hat-developer-hub/backstage-plugin-lightspeed` package.
* You have access to the `packages/app/src/App.tsx` file.

.Procedure

. Import and register the Lightspeed modules in your `packages/app/src/App.tsx` file:
+
[source,typescript]
----
import { createApp } from '@backstage/frontend-defaults';
import {
  lightspeedFABModule,
  lightspeedTranslationsModule,
} from '@red-hat-developer-hub/backstage-plugin-lightspeed/alpha';

export default createApp({
  features: [
    appDrawerModule,  // Required for app drawer integration
    lightspeedFABModule,  // Provides the floating action button
    lightspeedTranslationsModule,  // Adds internationalization support
    signInModule,
    navModule,
    // Additional features
  ],
});
----

. Save the file and restart your {product} instance.

.Verification

. Open your {product} instance in a web browser.

> . Verify that the *Lightspeed floating action* button appears on the page.

Reviewer: ```suggestion
. Verify that the *Lightspeed* floating action button appears on the page.
```

> . Click the *Floating action button* to open the chatbot drawer.

Reviewer: I'm not sure if this is right.
It's more like - Click the Lightspeed floating action button to open the *Developer Lightspeed" page.

Could you please verify this with the engineer? The floating action button is not the name of the Floating action button we see. It's called Developer Lightspeed but when you move the cursor around the FAB, it says "Open Lightspeed".

Reviewer: Also, we call it "virtual assistant" in all our docs and not a chatbot. Only one of its UI is called a "chatbot". I would suggest you use the same term.

Reviewer 2: "It's called Developer Lightspeed but when you move the cursor around the FAB, it says "Open Lightspeed"." is correct. Clicking on the Lightspeed button opens the last saved display mode, which can be overlay, drawer or the full page(that opens in a dedicated route), or defaults to overlay mode. 
Can we say "Click the Lightspeed floating action button to open the *Developer Lightspeed*" ?

> . Click the *Lightspeed floating action* button to open *{ls-short}*.

Reviewer: ```suggestion
. Click the *Lightspeed* floating action button to open {ls-short}.
```

> . Confirm that the button icon changes from the Lightspeed icon to a close icon when the chatbot is active.

Reviewer: (I'm not sure if this really happens. Could you please verify)

Reviewer 2: This is correct. Attaching a screenshot for ref

<img width="1205" height="829" alt="Screenshot 2026-05-06 at 8 21 17 PM" src="https://github.com/user-attachments/assets/9686a1a0-55c5-446c-97c4-a8c0694baa20" />

> . Confirm that the button icon changes from the Lightspeed icon to a close icon when the chatbot is active.

Reviewer: ```suggestion
. Confirm that the button icon changes from the *Lightspeed* icon to a close icon when the chatbot is active.
```

> . Ask the chatbot a question to verify it responds with AI-generated answers.

Reviewer: ```suggestion
. Enter a question to confirm that the virtual assistant responds with AI-generated answers.
```

> . Click the *Close* icon to close the chatbot drawer.

Reviewer: (I'm not sure we call it a "chatbot drawer")

Reviewer 2: We could just say "chatbot" here

cc @jmagak
